lib/location/scan_cellular.c

```diff
@@ -17,10 +17,22 @@
 LOG_MODULE_DECLARE(location, CONFIG_LOCATION_LOG_LEVEL);
 
 static struct lte_lc_ncell neighbor_cells[CONFIG_LTE_NEIGHBOR_CELLS_MAX];
+static struct lte_lc_cell gci_cells[CONFIG_LTE_NEIGHBOR_CELLS_MAX];
 static struct lte_lc_cells_info scan_cellular_info = {
-	.neighbor_cells = neighbor_cells
+	.neighbor_cells = neighbor_cells,
+	.gci_cells = gci_cells
 };
-static struct k_sem *scan_cellular_ready;
+
+/* Indicates when whole cellular scan operation is completed. This is given by caller. */
```

Comment:
"given" could be replaced with a different word here to avoid confusion, because we're talking about a semaphore which is given/taken. It could be for example "owned".